Adding ability to only allow adding on given times

Antonio-Laguna opened the pull request.

Commit:
Adding ability to only allow adding on given times

Commit:
Changing usage

Commit:
Fixing example

jgallen23 commented:
took a look.  found these issues:
- [x] can't select a time that ends at the whitelisted time (4:30 - 6:00 in the example)
- [x] saturday (day after tomorrow) shows available, but I can't click on it
- [x] unavailable should override today's color
- [x] I get an error when trying to resize $.declare.isWithinWhitelistedRange 1331

Commit:
Offseting time by one minute on template so it's

not needed to adjust the start/end whitelist period

Commit:
Some fixes

Ensuring today's color gets overwritten if time is unavailable.
Reusing canAdd instead of old method's signature

Commit:
Ensuring preview doesn't appear on hovering event

Commit:
Disabling preview on unavailable times/days
